src/main/java/com/iexec/common/task/TaskDescription.java

```diff
@@ -40,42 +41,93 @@
     private String callback;
     private DappType appType;
     private String appUri;
+    private String appFingerprint;
     private String cmd;
     private long maxExecutionTime;
     private boolean isTeeTask;
     private int botIndex;
     private int botSize;
     private int botFirstIndex;
     private boolean developerLoggerEnabled;
+    private String datasetAddress;
     private String datasetUri;
     private String datasetName;
     private String datasetChecksum;
     private List<String> inputFiles;
-    private boolean isCallbackRequested;
     private boolean isResultEncryption;
     private String resultStorageProvider;
     private String resultStorageProxy;
     private String teePostComputeImage;
     private String teePostComputeFingerprint;
 
+    /**
+     * Check if this task includes a dataset or not. The task is considered
+     * as including a dataset only if all fields of the dataset are non-empty,
+     * non-null values. The stack should ignore datasets with missing
+     * information since they, inevitably, break the workflow. In the case
+     * where those datasets are ignored, the worker will contribute an
+     * application error caused by the missing dataset file.
+     * 
+     * @return true if all dataset fields are all non-null,
+     * non-empty values, false otherwise.
+     */
+    public boolean containsDataset() {
+        return !StringUtils.isEmpty(datasetAddress) &&
```

Comment:
No need to add  !equals(EMPTY_ADDRESS) ?

Reply:
It is already handled [here](https://github.com/iExecBlockchainComputing/iexec-common/pull/238/files#diff-97c4da1cbc899dfa86abb540221a1acaad5f012e2454c0618fc6542a0171698aR68). But I'll add here also.